File: 2015/AI/AIClient_JAVA/src/event/MoveCharacterEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** MoveCharacter.java
 ** Event call to move a character
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import java.awt.Point;

public class MoveCharacterEvent extends OutgoingEvent {
	private int characterID;
	private Point position;

	public MoveCharacterEvent(int characterID, Point position) {
		this.characterID = characterID;
		this.position = position;
	}

	@Override
	public String toString() {
		String message = "Game:Move" + SEPARATOR;
		message += characterID + SEPARATOR;
		message += position.x + SEPARATOR;
		message += position.y;
		return message;
	}
}

File: 2015/AI/AIClient_Python/src/event/MoveCharacterEvent.py
'''
Created on Dec 20, 2014

@author: scarriere
'''
from event.OutgoingEvent import OutgoingEvent

class MoveCharacterEvent(OutgoingEvent):

    def __init__(self, characterId, positionX, positionY):
        self.characterId = characterId
        self.positionX = positionX
        self.positionY = positionY
        
    def toString(self):
        message = "Game:Move:"
        
        message += str(self.characterId) + self.SEPARATOR
        message += str(self.positionX) + self.SEPARATOR
        message += str(self.positionY)
        
        return message